examples/webrtc-direct/test.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const path = require('path')
+const execa = require('execa')
+const pDefer = require('p-defer')
+const uint8ArrayToString = require('uint8arrays/to-string')
+
+function startProcess (name, args = []) {
+    return execa('node', [path.join(__dirname, name), ...args], {
+        cwd: path.resolve(__dirname),
+        all: true
+    })
+}
+
+async function test () {
+    let listenerOutput = ''
+    let listenerAddr
+
+    const listenerProcReady = pDefer()
+
+    // Step 1, listener process
+    process.stdout.write('listener.js\n')
+
+    const listenerProc = startProcess('listener.js')
+    listenerProc.all.on('data', async (data) => {
+        process.stdout.write(data)
+
+        listenerOutput += uint8ArrayToString(data)
+        if (listenerOutput.includes('Listening on:') && listenerOutput.includes('/p2p/')) {
+            listenerAddr = listenerOutput.trim().split('Listening on:\n')[1].split('\n')[0]
+            //TODO What is the recommended way to paste `listenerAddr` into the `dialer.js`?
```

Comment:
What would be a recommended way to paste `listenerAddr` into `dialer.js`

Reply:
yes, there is no good way of doing this...

My suggestion is to allow the listener to receive a peerId via a parameter. Then, we can have the dialer browser with the default bootstrap id to be that one.

You can install `peer-id` module globally and do:

```sh
peer-id --type=ed25519 > id.json
```

This `id.json` file would also be commited in this example. Then, you can run the listener process providing a `peerId` file as a parameter like: https://github.com/libp2p/js-libp2p-relay-server/blob/v0.1.3/src/bin.js#L41

Another idea is really make this part of the example. So, use this `peer-id` file by default in the example and people can provide their own to override it. This would have the benefit that everything would just work and users would not need to change the dialer code to try this out. However, we need to make clear in the README that we are using a pregenerated peer-id by default. What do you think on this?